lib/kafka-consumer.js

```diff
@@ -134,9 +156,13 @@ Consumer.prototype._onWrapper = function (consumerInstance, eventName, cb) {
  * @param {Object} message The raw message.
  * @return {Object} The deserialized object.
  */
-Consumer.prototype.deserialize = function (type, message) {
+Consumer.prototype.deserialize = function (type, message, isKey) {
   try {
-    var parsed = magicByte.fromMessageBuffer(type, message.value, this.sr);
+    var parsed = magicByte.fromMessageBuffer(
+      type,
+      !isKey ? message.value : message.key,
```

Comment:
Instead of `!isKey` you can use a strict comparison to avoid coertion problems - `isKey === true ? message.key : message.value`